### ISSUE TITLE
Package dbt w/ dbt-snowflake adapter fails to build.

### ISSUE THREAD

**Issue by jomccr** [NixOS/nixpkgs#263987]
I am trying to build a system with dbt installed with the dbt-snowflake adapter as shown here 

https://github.com/NixOS/nixpkgs/blame/63678e9f3d3afecfeafa0acead6239cdb447574c/pkgs/development/python-modules/dbt-core/default.nix#L101-L106

When I run the following
```
nix-shell -p 'dbt.withAdapters (adapters: [adapters.dbt-snowflake])'
```
I get an error
```
ModuleNotFoundError: No module named 'dbt_semantic_interfaces'
error: builder for '/nix/store/4rwm8qj6xg927pmgcklijdrdn7yq2rir-python3.11-dbt-snowflake-1.6.2.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/2gsds8glfpkbwgirdg29aiczkgy2ij11-python3-3.11.5-env.drv' failed to build
```

This bug also happens when I try to install `dbt` standalone without any adapters.

```
bash-5.2# nix-shell -p 'dbt'

[nix-shell:/]# dbt --version
Traceback (most recent call last):
  File "/nix/store/m4frwhw7wzpbkzzja2vj89x71r5wyq2j-python3.11-dbt-core-1.6.2/bin/.dbt-wrapped", line 6, in <module>
    from dbt.cli.main import cli
  File "/nix/store/m4frwhw7wzpbkzzja2vj89x71r5wyq2j-python3.11-dbt-core-1.6.2/lib/python3.11/site-packages/dbt/cli/__init__.py", line 1, in <module>
    from .main import cli as dbt_cli  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/m4frwhw7wzpbkzzja2vj89x71r5wyq2j-python3.11-dbt-core-1.6.2/lib/python3.11/site-packages/dbt/cli/main.py", line 13, in <module>
    from dbt.cli import requires, params as p
  File "/nix/store/m4frwhw7wzpbkzzja2vj89x71r5wyq2j-python3.11-dbt-core-1.6.2/lib/python3.11/site-packages/dbt/cli/requires.py", line 3, in <module>
    from dbt.adapters.factory import adapter_management, register_adapter
  File "/nix/store/m4frwhw7wzpbkzzja2vj89x71r5wyq2j-python3.11-dbt-core-1.6.2/lib/python3.11/site-packages/dbt/adapters/factory.py", line 8, in <module>
    from dbt.adapters.base.plugin import AdapterPlugin
  File "/nix/store/m4frwhw7wzpbkzzja2vj89x71r5wyq2j-python3.11-dbt-core-1.6.2/lib/python3.11/site-packages/dbt/adapters/base/__init__.py", line 6, in <module>
    from dbt.adapters.base.connections import BaseConnectionManager  # noqa: F401
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/m4frwhw7wzpbkzzja2vj89x71r5wyq2j-python3.11-dbt-core-1.6.2/lib/python3.11/site-packages/dbt/adapters/base/connections.py", line 35, in <module>
    from dbt.contracts.graph.manifest import Manifest
  File "/nix/store/m4frwhw7wzpbkzzja2vj89x71r5wyq2j-python3.11-dbt-core-1.6.2/lib/python3.11/site-packages/dbt/contracts/graph/manifest.py", line 27, in <module>
    from dbt.contracts.graph.nodes import (
  File "/nix/store/m4frwhw7wzpbkzzja2vj89x71r5wyq2j-python3.11-dbt-core-1.6.2/lib/python3.11/site-packages/dbt/contracts/graph/nodes.py", line 15, in <module>
    from dbt.contracts.graph.semantic_models import (
  File "/nix/store/m4frwhw7wzpbkzzja2vj89x71r5wyq2j-python3.11-dbt-core-1.6.2/lib/python3.11/site-packages/dbt/contracts/graph/semantic_models.py", line 3, in <module>
    from dbt_semantic_interfaces.references import (
ModuleNotFoundError: No module named 'dbt_semantic_interfaces'
```

cc @tjni @mausch

**Comment by jomccr** [comment]
Below is the Dockerfile I'm using to set up the environment where I run these commands. 

```Dockerfile
FROM nixos/nix
RUN nix-channel --update
RUN nix-build -A hello '<nixpkgs>'
```
I'm building and running with the following command
```zsh
 docker build -t testnix -f ./Dockerfile . && docker run -it testnix
```

**Comment by austinbutler** [comment]
In the meantime you can use nixpkgs revision dfcffbd74fd6f0419370d8240e445252a39f4d10 to get an older by seemingly working dbt.

**Comment by a-n-n-a-l-e-e** [comment]
should be fixed with PR #268081
